google/data_source_google_kms_key_ring_test.go

```diff
@@ -0,0 +1,65 @@
+package google
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+func TestAccDataSourceGoogleKmsKeyRing_basic(t *testing.T) {
+	projectId := "terraform-" + acctest.RandString(10)
+	projectOrg := getTestOrgFromEnv(t)
+	billingAccount := getTestBillingAccountFromEnv(t)
+	keyRingName := fmt.Sprintf("tf-test-%s", acctest.RandString(10))
+	folderId := os.Getenv("GOOGLE_FOLDER_ID")
```

Comment:
We don't currently have a pattern for running tests in many configurations based on environment variables. If we need test 2 different configurations of a resource we have to create 2 separate tests. So as this test is encoded it will only ever run 1 of these paths in our CI system. Can you remove the logic for the folder test as it will be dead code for our CI system and anybody who doesn't have that variable set?

Normally I would encourage 2 separate tests, but in this case I think testing both project configurations is a more of a `google_project` test than a KMS data source test.